### ISSUE TITLE
Create HTML sitemap page

### ISSUE THREAD

**Issue by codemonkey800** [chanzuckerberg/napari-hub#899]
An HTML sitemap page will be an additional page on the napari hub that'll include every possible link in the napari hub. Since this will be a new page, we'll need to:

- Create new Next.js route for the sitemap page
- Have designs for new HTML sitemap page
- Implement UI for sitemap page

The sitemap page should include all of the static pages like the home, markdown, and collection home pages. It should also dynamically have entries for every possible plugin and collection available on the hub.

**Comment by EricMaxWang** [comment]
Looping in @Janeece as this will require some design effort. @codemonkey800 to follow up on specifics.

**Comment by Janeece** [comment]
@codemonkey800 @EricMaxWang I can provide design recommendations for (1) HTML sitemap page and (2) where to add a navigational link to the sitemap by end of week.

**Comment by EricMaxWang** [comment]
@Janeece Probably best to post it here and then have @codemonkey800 incorporate it into the description above!

**Comment by Janeece** [comment]
> The sitemap page should include all of the static pages like the home, markdown, and collection home pages. It should also dynamically have entries for every possible plugin and collection available on the hub.

@codemonkey800 can you clarify for me what page you would consider "markdown" pages? (I'm guessing FAQ/about/policy, etc but just want to confirm)

**Comment by Janeece** [comment]
Hi @codemonkey800 @EricMaxWang here's my design recommendation for the HTML sitemap. Please let me know if you have any questions or feedback on these designs! Quick note, things i mark as "nice to have" are non-critical suggestions, can can be cut if scope doesn't allow. 

**Design Recommendations:**
- Add "Sitemap" link to footer. 
- Created lowest-lift sitemap layout with essential links and hierarchy
- Explored 2 "nice to have" functional additions to help users navigate/find plugins, but I'm concerned about scope for design and eng: 
  - opt 1:   search for plugin section to filter down plugins and help users find plugins with appended names
  - opt 2: paginate plugin section by letter groupings to cap # of plugin links

**Feedback Requested:**
- any questions or refinements needed
- Do we want to include plugin section search or pagination in scope? These will need additional design work. 

**View Design in Figma:** https://www.figma.com/file/G3r9ATsRj3G3sxTt8B5qi1/SEO-HTML-Sitemap?node-id=0%3A1&t=IcMBIlQM3IZ3tlw1-1

**Screenshots:**
<img width="691" alt="Screenshot 2023-02-17 at 11 29 53 AM" src="https://user-images.githubusercontent.com/63013360/219774975-302428b3-0573-4255-9ebe-c0258ce0da99.png">
<img width="261" alt="Screenshot 2023-02-17 at 11 30 01 AM" src="https://user-images.githubusercontent.com/63013360/219775027-129ae51e-499e-43e2-a868-43898f165fd0.png">
<img width="754" alt="Screenshot 2023-02-17 at 11 33 28 AM" src="https://user-images.githubusercontent.com/63013360/219775895-c7d3705a-ae08-40c7-9d4f-f182ed11bdc0.png">

Nice to have opt 1:
<img width="799" alt="Screenshot 2023-02-17 at 11 46 11 AM" src="https://user-images.githubusercontent.com/63013360/219775462-3129c75c-a218-46a8-8091-4b8fb76c34ba.png">
Nice to have opt 2:
<img width="798" alt="Screenshot 2023-02-17 at 11 46 20 AM" src="https://user-images.githubusercontent.com/63013360/219775506-e24511d2-97b5-4d17-a742-0684d5c7a245.png">

**Comment by EricMaxWang** [comment]
Thanks for the the heads up, @Janeece!

Tagging @neuromusic for decision on Opt 1 vs. Opt 2 design look and feel.

**Comment by neuromusic** [comment]
> functional additions to help users navigate/find plugins

The primary motivation for this page is to support web crawlers (we have the primary search and exploration functionality to help users navigate and find plugins, so I'm not too concerned about meeting those needs here.

If the number of plugins makes this page burdensome & increases load times too much, then we can consider Opt 2 perhaps? I fear that Opt 1 (adding a plugin search) would be too duplicative with the main plugin search functionality.

TL;DR: my vote is to keep scope restricted to "lowest lift" layout

**Comment by Janeece** [comment]
@neuromusic I think that makes a lot of sense and I agree!

**Comment by Janeece** [comment]
Posting for visibility in this issue, specs and responsive behavior has been added to the figma file:
https://www.figma.com/file/G3r9ATsRj3G3sxTt8B5qi1/SEO----HTML-Sitemap?node-id=1%3A3&t=6pgi3AeHMyz3BXJl-1

cc @codemonkey800
